### PR TITLE
LPS-109799 Adding the cancel button in edit form instance record screen

### PR DESCRIPTION
**Pull request by marcosapmf**
Related issue: https://issues.liferay.com/browse/LPS-109799

### CODE REVIEW

**Comment by natocesarrego** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance_record.jsp [hunk 1]
```diff
@@ -29,15 +29,16 @@ String title = ParamUtil.getString(request, "title");
 renderResponse.setTitle(GetterUtil.get(title, LanguageUtil.get(request, "view-form")));
 %>
 
-<div class="portlet-forms">
-	<div class="ddm-form-basic-info ddm-form-success-page">
-		<div class="container-fluid-1280">
-			<h1 class="ddm-form-name"><%= ddmFormAdminDisplayContext.getFormName() %></h1>
+<div class="container-fluid-1280 ddm-form-builder-app editing-form-entry">
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance_record.jsp:32]: @marcosapmf, where is the `</div>`? Did you try to move this `div` to include the form name and description elements? I'm not sure if I got what's the point of this modification, can you please clarify?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-uad/src/main/java/com/liferay/dynamic/data/mapping/uad/display/DDMFormInstanceRecordUADDisplay.java [hunk 2]
```diff
@@ -97,9 +97,6 @@ public String getEditURL(
 			).put(
 				portletNamespace.concat("mvcPath"),
 				new String[] {"/admin/edit_form_instance_record.jsp"}
-			).put(
-				portletNamespace.concat("readOnly"),
-				new String[] {Boolean.FALSE.toString()}
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-uad/src/main/java/com/liferay/dynamic/data/mapping/uad/display/DDMFormInstanceRecordUADDisplay.java:102]: @marcosapmf, just to make more explicit to the next pull request reviewer, I believe that would be better move this modification to a separate commit and explain it through the commit message, can you please do this?

**Comment by natocesarrego** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/asset/model/DDMFormAssetRenderer.java [hunk 3]
```diff
@@ -152,6 +157,38 @@ public String getTitle(Locale locale) {
...
+	public PortletURL getURLEdit(
+			LiferayPortletRequest liferayPortletRequest,
+			LiferayPortletResponse liferayPortletResponse)
+		throws Exception {
+
+		String portletNamespace =
+			DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM_ADMIN;
+
+		HttpServletRequest httpServletRequest =
+			liferayPortletRequest.getHttpServletRequest();
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/asset/model/DDMFormAssetRenderer.java:170]: @marcosapmf, as used rule. `httpServletRequest` is used first then `portletNamespace`.

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp [hunk 4]
```diff
@@ -191,6 +191,18 @@ renderResponse.setTitle(headerTitle);
...
+									%>
+
+									<c:if test="<%= Validator.isNotNull(editURL) %>">
+										<liferay-frontend:management-bar-button
+											href="<%= assetRenderer.getURLEdit(liferayPortletRequest, liferayPortletResponse).toString() %>"
+											icon="pencil"
+											label="edit"
+										/>
+									</c:if>
+
```
Comment (natocesarrego) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp:205]: @marcosapmf, do we need to add this code? Isn't it already available in some place in the code? After all, the other Portlets using Workflow which allow the asset edit already display the edit icon, right?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java [hunk 5]
```diff
@@ -125,8 +131,11 @@ protected DDMFormRenderingContext createDDMFormRenderingContext(
 		DDMFormRenderingContext formRenderingContext =
 			new DDMFormRenderingContext();
 
-		formRenderingContext.setHttpServletRequest(
-			_ddmFormAdminRequestHelper.getRequest());
+		HttpServletRequest httpServletRequest =
+			_ddmFormAdminRequestHelper.getRequest();
+
+		formRenderingContext.setHttpServletRequest(httpServletRequest);
+
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java:138]: @marcosapmf, why was is necessary?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance_record.jsp (hunk 1 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/admin/edit_form_instance_record.jsp:32]: @natocesarrego sorry, this `<div>` is duplicated, it should be caused by rebase that I did. I'm going to fix it :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/asset/model/DDMFormAssetRenderer.java (hunk 3 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/asset/model/DDMFormAssetRenderer.java:170]: it's necessary to avoid a NPE [here](https://github.com/liferay/liferay-portal/blob/2152cecbba72d32572fa776486d7ea59bb0ce585/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryImpl.java#L126)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java (hunk 5 above)
Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java:138]: it's necessary to avoid a NPE [here](https://github.com/liferay/liferay-portal/blob/2152cecbba72d32572fa776486d7ea59bb0ce585/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/java/com/liferay/dynamic/data/mapping/form/renderer/internal/DDMFormTemplateContextFactoryImpl.java#L126)

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp (hunk 4 above)
Comment (marcosapmf) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/edit_workflow_task.jsp:205]: @natocesarrego right, I am going do check more once :)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java (hunk 5 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormViewFormInstanceRecordDisplayContext.java:138]: To avoid the NPE then the variable `httpServletRequest` should be used there or some check should be performed there too, right? Can you please give more details, @marcosapmf? Thanks.
